code/modules/mob/living/brain/brain_say.dm

```diff
@@ -1,14 +1,25 @@
-/mob/living/brain/say(message, bubble_type, list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null, filterpoof = null, message_range = 7, datum/saymode/saymode = null)
-	if(!(container && istype(container, /obj/item/mmi)))
-		return //No MMI, can't speak, bucko./N
-	else
-		if(prob(emp_damage*4))
-			if(prob(10))//10% chane to drop the message entirely
-				return
-			else
-				message = Gibberish(message, emp_damage >= 12)//scrambles the message, gets worse when emp_damage is higher
+/mob/living/brain/say(
+	message,
+	bubble_type,
+	list/spans = list(),
+	sanitize = TRUE,
+	datum/language/language,
+	ignore_spam = FALSE,
+	forced,
+	filterproof = FALSE,
+	message_range = 7,
+	datum/saymode/saymode,
+	list/message_mods = list(),
+)
+	if(prob(emp_damage * 4))
+		if(prob(10))//10% chane to drop the message entirely
+			return
+		message = Gibberish(message, emp_damage >= 12)//scrambles the message, gets worse when emp_damage is higher
+
+	return ..()
 
-		..()
+/mob/living/brain/can_speak(allow_mimes)
+	return ..() && istype(container, /obj/item/mmi)
```

Comment:
i think we should also flip the parent call and the `istype()` here as well for short-circuiting.

Reply:
```suggestion
	return istype(container, /obj/item/mmi) && ..()
```